packages/ckeditor5-html-support/src/htmlcomment.js

```diff
@@ -89,4 +89,120 @@ export default class HtmlComment extends Plugin {
 			return true;
 		} );
 	}
+
+	/**
+	 * Creates an HTML comment on the specified position and returns its ID.
+	 *
+	 * *Note*: If two comments are created at the same position, the second comment will be inserted before the first one.
+	 *
+	 * @param {module:engine/model/position~Position} position
+	 * @param {String} content
+	 * @returns {String} Comment ID. This ID can be later used to e.g. remove the comment from the content.
+	 */
+	createHtmlComment( position, content ) {
+		const id = uid();
+		const editor = this.editor;
+		const model = editor.model;
+		const root = model.document.getRoot();
+		const markerName = `$comment:${ id }`;
+
+		return model.change( writer => {
+			const range = writer.createRange( position );
+
+			writer.addMarker( markerName, {
+				usingOperation: true,
+				affectsData: true,
+				range
+			} );
+
+			writer.setAttribute( markerName, content, root );
+
+			return markerName;
+		} );
+	}
+
+	/**
+	 * Removes an HTML comment with the given comment ID.
+	 *
+	 * It throws an error if the comment with the given ID does not exist.
+	 * Note that a comment can be removed also by removing the content in which it is created.
+	 *
+	 * @param {String} commentID The ID of the comment to be removed.
+	 */
+	removeHtmlComment( commentID ) {
+		const editor = this.editor;
+		const root = editor.model.document.getRoot();
+
+		const marker = editor.model.markers.get( commentID );
+
+		if ( !marker ) {
+			/**
+			 * An HTML comment with the given ID does not exist.
+			 *
+			 * @error html-comment-does-not-exist
+			 */
+			throw new CKEditorError( 'html-comment-does-not-exist', null );
+		}
```

Comment:
> I wonder why trying to remove a non-existent comment ends with an exception, but trying to read the content of a non-existent comment only returns `null`. Maybe it should be unified?
> 
> Personally, I would do as it is in `MarkerCollection`, i.e. the method for removing a comment should return `true`, when the comment has been found and removed, or `false` otherwise. I think that throwing an exception is not needed here, because when you try to remove a non-existent comment, the editor state is the same as the one expected by the user, i.e. the comment to be removed does not exist anyway, so the user's expectation is fulfilled.

The problem here is the fact that something can be removed either via the API or via the editor. Maybe the current state isn't the best one though. Probably with a `doesHtmlCommentExist( commentID )` both of the above methods could throw errors if the comment does not exist

Reply:
After a f2f talk, I changed it so these methods do not throw errors and the `removeHtmlComment` returns `true/false` according to the presence of the removed comment.